### DISCUSSION TITLE
Outgoing mail activity - how to show public IP addresses from where the messages are opened

### DISCUSSION THREAD

**Discussion by Ascar699** [postalserver/postal#1850] [Help with using Postal] [answered]
Postal team and everyone who know,

I am running postal in an Ubuntu VM behind pfSense with HAProxy.  Sending works fine and Postal captures message stats like opens and clicks.  However all open and click events show as happening on 192.168.2.254 which is the IP address assigned to the LAN interface of my firewall.  I made an extensive research on how to force the HAProxy module on pfSense firewall pass the public IP address to the Postal web server but nothing worked.  I tried setting Use 'forwardfor' option in the frontend and adding 'option forwardfor' in the backend.  I was able to fix the similar issue for the Apache-based webservers by leaving all HAProxy setting alone and modifying the apache2.conf file by adding 'RemoteIPHeader X-Forwarded-For' string and altering the log file format.

My question - can I do similar modification to the webserver in my Postal container?  How do I do that?  Any suggestion and advise is appreciated.

**Comment by willpower232** [comment, marked as answer]
This has been pointed out in #1816, I don't believe it has been resolved yet.

**Comment by Ascar699** [comment]
Thank you for admitting the issue.  Waiting for resolution.  I believe this issue does not exist when Postal is deployed with Caddy.

**Reply by willpower232** [reply]
in this case, the pfsense is the issue as people connecting to Postal go via the extra device which creates the forwarded-for header that Postal is not using apparently. If you used caddy and pfsense, you'd still have a problem probably, I'm not sure how pfsense works

**Reply by Ascar699** [reply]
No, Will.  I meant a different things.  Before I made Postal work for me I installed it several times, thanks to virtualization technologies like snapshots, I did not have to start from every time from scratch.  One scenario that worked well for me was when I assigned a public IP address to the LAN port of the VM and used Caddy - everything worked fine including receiving IP addresses of the message recipients who opened messages and clicked.  In this scenario the traffic was not traversing pfSense and the VM was exposed to the internet. 

However I did not want to waste a whole IP address to a single Postal task and also did not feel peaceful about Postal VM being directly connected to the public internet.  So, I spent a while before I figured how to make Postal work from behind pfSense with HAProxy.

The issue with detection of recipient's IP address only appears when my HAProxy on pfSense forwards traffic to my Postal VM and it seems that it does not matter whether Caddy is used in this case.  In my deployment I don't have any certificates on the Postal VM and the SSL part is taken care of by HAProxy and while it takes care of web session encryption the 'forwardfor' feature does not work.  This is what I meant.

I also believe there are some other things that don't work as expected when Postal is behind HAProxy.
